src/Domain/Bootstrap.php

```diff
@@ -130,17 +130,21 @@ function() {
 	 * Define the global feature flag.
 	 */
 	protected function define_feature_flag() {
-		$allowed_flags = [ 'experimental', 'stable' ];
-		$flag          = getenv( 'WOOCOMMERCE_BLOCKS_PHASE' );
-		if ( ! in_array( $flag, $allowed_flags, true ) ) {
-			if ( file_exists( __DIR__ . '/../../blocks.ini' ) ) {
-				$woo_options = parse_ini_file( __DIR__ . '/../../blocks.ini' );
-				$flag        = is_array( $woo_options ) && 'experimental' === $woo_options['woocommerce_blocks_phase'] ? 'experimental' : 'stable';
-			} else {
-				$flag = 'stable';
-			}
-		}
-		define( 'WOOCOMMERCE_BLOCKS_PHASE', $flag );
+		add_action(
+			'plugins_loaded',
```

Comment:
What are we waiting for exactly? I don't think we need to wait - we just need to see if a global/constant is defined by this point and derive our context.

Reply:
we need to wait for the code `woocommerce-gutenberg-products-block.php` to run, since this file runs before it, alternatively we can the code in `woocommerce-gutenberg-products-block.php` at priority 9 and run this without a hook

Reply:
or even further avoid hooks all together as you said, took me some time to understand it 😅